storagenode/nodestats/cache.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package nodestats
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"storj.io/storj/internal/date"
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/storagenode/reputation"
+	"storj.io/storj/storagenode/storageusage"
+	"storj.io/storj/storagenode/trust"
+)
+
+var (
+	// NodeStatsCacheErr defines node stats cache loop error
+	NodeStatsCacheErr = errs.Class("node stats cache error")
+)
+
+// Cache runs cache loop and stores reputation stats
+// and storage usage into db
+type Cache struct {
+	log *zap.Logger
+
+	service        *Service
+	trust          *trust.Pool
+	reputationDB   reputation.DB
+	storageusageDB storageusage.DB
+
+	reputationCycle *sync2.Cycle
+	storageCycle    *sync2.Cycle
```

Comment:
The cycles don't have to be pointers here.

Reply:
In other places like `sender` pointer is used for cycle, also constructor returns pointer

Reply:
Mostly it should be non-pointers. You can initialize it with:

```
reputationCycle: *sync2.NewCycle ...
```

Reply:
Ok, just thought that returning it as a pointer is a hint of how it should be used